srv/components/defaults.yaml

```diff
@@ -17,122 +17,145 @@
     # FIXME EOS-12508 not considered yet ?
     {% set system_repo = '/'.join([base_bundle_url, 'centos7.7']) %}
   {% endif %}
 
-  {% set cortx_repo = '/'.join([base_bundle_url, 'cortx_iso']) %}
-  {% set cortx_deps_repo = '/'.join([base_bundle_url, '3rd_party']) %}
-  {% set epel_repo = '/'.join([cortx_deps_repo, 'EPEL-7']) %}
-  {% set saltstack_repo = '/'.join([cortx_deps_repo, 'commons', 'saltstack-3001']) %}
-  {% set glusterfs_repo = '/'.join([cortx_deps_repo, 'commons', 'glusterfs']) %}
+  {% set release_repo = '/'.join([base_bundle_url, 'cortx_iso']) %}
+  {% set deps_bundle_url = '/'.join([base_bundle_url, '3rd_party']) %}
 
-  {% set repo_url = cortx_repo %}
-  {% set cortx_commons_url = cortx_deps_repo %}
-
-base_repos:
-  centos_repos: [] # FIXME redhat with subscription disabled
-  #  - id: "cortx_platform_base"
-  #    url: {{ salt["environ.get"]('CORTX_PLATFORM_BASE', system_repo) }}
-  epel_repo:
-    id: "epel"
-    url: {{ salt["environ.get"]('EPEL', epel_repo) }}
-  saltstack_repo:
-    id: "saltstack"
-    url: {{ salt["environ.get"]('SALTSTACK', saltstack_repo) }}
+{% elif not deps_bundle_url %}
+  # deps bundle url implies that we follow new repo structure
 
+  # LEGACY REPOS STRUCTURE
 
-{% else %}
+  {% set cortx_platform_base = salt["environ.get"]('CORTX_PLATFORM_BASE', "http://ssc-satellite1.colo.seagate.com/pulp/repos/EOS/Library/custom/CentOS-7/CentOS-7-OS/") %}
+    # url: http://mirrors.usc.edu/pub/linux/distributions/centos/7.7.1908/os/x86_64/
+  {% set cortx_platform_extras = salt["environ.get"]('CORTX_PLATFORM_EXTRAS', "http://ssc-satellite1.colo.seagate.com/pulp/repos/EOS/Library/custom/CentOS-7/CentOS-7-Extras/") %}
+    # url: http://mirrors.usc.edu/pub/linux/distributions/centos/7.7.1908/extras/x86_64/
+  {% set cortx_platform_updates = salt["environ.get"]('CORTX_PLATFORM_UPDATES', "http://ssc-satellite1.colo.seagate.com/pulp/repos/EOS/Library/custom/CentOS-7/CentOS-7-Updates/") %}
+    # url: http://mirrors.usc.edu/pub/linux/distributions/centos/7.7.1908/updates/x86_64/
 
-  {% set repo_url = pillar['release']['target_build'] %}
+  {% set os_repos = [
+        ["id": "cortx_platform_base", "url": os_base_repo],
+        ["id": "cortx_platform_extras", "url": os_extras_repo],
+        ["id": "cortx_platform_updates", "url": os_updates_repo]
+      ]
+  %}
 
   {% if "RedHat" in grains['os'] %}
-    {% set cortx_commons_url = pillar['commons']['cortx_commons']['RedHat'] %}
+    {% set commons_repo = pillar['commons']['cortx_commons']['RedHat'] %}
   {% else %}
-    {% set cortx_commons_url = pillar['commons']['cortx_commons']['CentOS'] %}
+    {% set commons_repo = pillar['commons']['cortx_commons']['CentOS'] %}
   {% endif %}
 
+  {% set epel_repo = "http://ssc-satellite1.colo.seagate.com/pulp/repos/EOS/Library/custom/EPEL-7/EPEL-7/" %}
+    # url: https://dl.fedoraproject.org/pub/epel/7/x86_64/
+  {% set saltstack_repo = "https://repo.saltstack.com/py3/redhat/$releasever/$basearch/3001" %}
+
+{% endif %}
```

Comment:
In the past I have seen a JINJA without a closing else after an if/elif causes minion to return false.
Not sure it would happen the same way in this case, but there's no harm in addition of an empty else block with a blank or default statement

Reply:
Done